autosubmit/notifications/mail_notifier.py

```diff
@@ -20,59 +20,111 @@
 import smtplib
 import email.utils
 from email.mime.text import MIMEText
+from email.mime.multipart import MIMEMultipart
+from email.mime.application import MIMEApplication
 from log.log import Log
+from pathlib import Path
+from autosubmitconfigparser.config.basicconfig import BasicConfig
+from typing import List, TYPE_CHECKING
+
+
+if TYPE_CHECKING:
+    from autosubmit.platforms.platform import Platform
+
 
 class MailNotifier:
     def __init__(self, basic_config):
         self.config = basic_config
 
-    def notify_experiment_status(self, exp_id,mail_to,platform):
-        message_text = self._generate_message_experiment_status(exp_id, platform)
-        message = MIMEText(message_text)
-        message['From'] = email.utils.formataddr(('Autosubmit', self.config.MAIL_FROM))
-        message['Subject'] = f'[Autosubmit] Warning a remote platform is malfunctioning'
+    def notify_experiment_status(
+            self,
+            exp_id: str,
+            mail_to: str,
```

Comment:
There's a type error in `test_mail.py`. I thought `mail_to` was `str`, but looks like it's a `List[str]`, sorry. Can you update it here?

![image](https://github.com/user-attachments/assets/a80f9fe7-5f3d-4eb4-a540-5ddad165c15f)